codegen/plugin.go

```diff
@@ -0,0 +1,48 @@
+package codegen
+
+import (
+	"github.com/vektah/gqlparser/ast"
+)
+
+type Plugin interface {
```

Comment:
Do we want one large interface with lots of methods or lots of small interfaces that can optionally be implemented?

eg we may eventually have codegen hooks here, but most plugins wont use them.

Reply:
lots of 1-function interfaces will be easier to use, then one big.